tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+
+    # return a testinfra connection to the container
+    yield testinfra.get_host("podman://" + name)
+
+@pytest.fixture(scope='session')
+def smoker():
+    return testinfra.get_host("local://localhost")
+
+@pytest.fixture
+def server_hostname(variables):
+    assert variables['server_hostname']
+    return variables['server_hostname']
+
+@pytest.fixture
+def foreman_version(variables):
+    assert variables['foreman_version']
+    return variables['foreman_version']
+
+def pytest_generate_tests(metafunc):
+    with open(metafunc.config.option.variables[0]) as file:
+      variables = json.load(file)
+      metafunc.parametrize('client', variables['clients'], indirect=True, scope='session')
+
+def teardown_module():
+    podman_ids = subprocess.check_output(['podman', 'ps', '-q', '--filter', "name=smoker-client"]).decode().strip()
+    podman_ids = podman_ids.split("\n")
+
+    for id in podman_ids:
+        subprocess.check_call(['podman', 'rm', '-f', id])
+
+
+@pytest.mark.katello_agent
+def test_install_subscription_manager(client, smoker):
+    client.run('yum -y install subscription-manager')
+    sub_man = client.package('subscription-manager')
+    assert sub_man.is_installed
+
+    hammer = smoker.package('tfm-rubygem-hammer_cli_katello')
+    assert hammer.is_installed
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_username_and_password(client, entities, server_hostname):
+    client.run("yum erase -y 'katello-ca-consumer-*'")
+
+    client.run_expect([0], "rpm -Uvh http://%s/pub/katello-ca-consumer-latest.noarch.rpm" % server_hostname)
+    client.run_expect([0], "subscription-manager register --force --org=%s --username=admin --password=changeme --env=Library" % entities['organization_label'])
+
+
+@pytest.mark.katello_agent
+def test_register_subscription_manager_with_activation_key(client, entities):
+    client.run_expect([0], "subscription-manager register --force --org=%s --activationkey=\"%s\"" % (entities['organization_label'], entities['activation_key']))
+
+    consumed = client.check_output("subscription-manager list --consumed")
+    assert entities['product'] in consumed
+
+
+@pytest.mark.katello_agent
+def test_host_is_registered(client, smoker):
+    hostname = client.check_output('hostname -s')
+    smoker.run_expect([0], "hammer host info --name %s" % hostname)
+
+
+@pytest.mark.katello_agent
+def test_enable_content_view_repo(client, entities):
+    enable = client.check_output("subscription-manager repos --enable=%s_%s_%s" % (entities['organization_label'], entities['product_label'], entities['yum_repository_label']))
+    assert "is enabled for this system" in enable
+
+
+def test_install_katello_host_tools(client, foreman_version):
+    release = client.system_info.release.split('.')[0]
+    client.run_expect([0], "yum -y install http://yum.theforeman.org/client/%s/el%s/x86_64/foreman-client-release.rpm" % (foreman_version, release))
```

Comment:
Need a better place to do this to help support other distributions and spins that have different client repositories.

Reply:
I'd suggest a `foreman_client_release_rpm_url` fixture.